docarray/array/mixins/plot.py

```diff
@@ -73,30 +120,12 @@ def summary(self):
         tables.append(table)
 
         all_attrs_names = tuple(sorted(all_attrs_names))
-        if all_attrs_names:
 
-            attr_table = Table(box=box.SIMPLE, title='Attributes Summary')
-            attr_table.add_column('Attribute')
-            attr_table.add_column('Data type')
-            attr_table.add_column('#Unique values')
-            attr_table.add_column('Has empty value')
-
-            all_attrs_values = self._get_attributes(*all_attrs_names)
-            if len(all_attrs_names) == 1:
-                all_attrs_values = [all_attrs_values]
-            for _a, _a_name in zip(all_attrs_values, all_attrs_names):
-                try:
-                    _a = set(_a)
-                except:
-                    pass  # intentional ignore as some fields are not hashable
-                _set_type_a = set(type(_aa).__name__ for _aa in _a)
-                attr_table.add_row(
-                    _a_name,
-                    str(tuple(_set_type_a)),
-                    str(len(_a)),
-                    str(any(_aa is None for _aa in _a)),
-                )
-            tables.append(attr_table)
+        if all_attrs_names:
+            if isinstance(self, DocumentArrayInMemory):
+                self._update_tables_exhaustive(tables, all_attrs_names)
+            else:
+                self._update_tables_information(tables, all_attrs_names)
```

Comment:
I believe we should avoid these checks and rely on inheritance and overriding methods (and if we apply the suggestion, the method should just return information and shouldn't handle the tables I guess)